lab4/context.tex

```diff
@@ -8,8 +8,55 @@ \subsection{Tech stack overview}
+\subsubsection{Functional requirements}
+
+\newlist{frlist}{enumerate}{1}
+\setlist[frlist]{label=\textbf{FR\arabic*.}, align=left, leftmargin=*}
+
+\begin{frlist}
+    \item System shall allow users to list quizzes.
+    \item System shall ensure that only logged in users can access quizzes.
+    \item System shall only allow creation, editing and deletion of the topic quizzes to owners of that topic.
+    \item System shall support users taking the quiz.
+    \item System shall check whether user answer is correct and display appropriate feedback.
+    \item System shall ensure that users cannot restart the quiz that was edited or deleted during the quiz session.
+    \item System shall allow topic owner to delete any quiz that belongs to their topic.
+    \item System shall allow topic owner to create the quiz in their topic.
+    \item System shall allow topic owner to edit any quiz in their topic.
+    \item System shall prompt for confirmation when deleting quizzes.
+    \item System shall support uploading image for questions.
+    \item System shall support setting quiz name.
+    \item System shall support adding, editing and deleting quiz questions and options.
+    New functional requirements:
+    \item System shall allow users to create a quiz room.
+    \item System shall generate a random code for the quiz room.
+    \item System shall allow the users to join a quiz room by entering the code.
+    \item System shall support taking quizzes with multiple participants.
+    \item System shall display feedback and their current position to all participants after answering a question.
+    \item System shall display the results of the quiz to all participants.
+    \item System shall allow a participant to leave a quiz at any time.
+    \item System shall support win streaks and achievements.
+    \item System shall support displaying the current win streak and achievements of the user. (Not sure about these two. Would be nice but perhaps require changing too many things and overall too much work)
+\end{frlist}
+
+
+\subsubsection{Nonfunctional requirements}
+
+\newlist{nfrlist}{enumerate}{1}
+\setlist[nfrlist]{label=\textbf{NFR\arabic*.}, align=left, leftmargin=*}
+
+\begin{nfrlist}
+    \item System shall support PNG, SVG, JPEG, WEBP, GIF question image formats.
+    \item System shall ensure that quiz functionality related pages are supported by common screenreaders (VoiceOver, Microsoft Narrator, Jaws).
+    \item System shall ensure that quiz functionality related pages are displayed on desktop and mobile without horizontal scrollbars or loss of information.
+    \item System shall support image upload and download speeds of at least 500Mb/s. (Changed this requirement as the average speed in a location is not under our control)
+    \item System shall support multiple participant quizzes.
+    \item System shall ensure that the quiz room code is unique.
+    \item System shall ensure that bot users cannot join the quiz room.
```

Comment:
All requirements must be covered by our design, and I am not sure we need that extra work. Also it's not exactly specific to be a non-functional requirement.